### DISCUSSION TITLE
Only use most recent direction input in osu!catch

### DISCUSSION THREAD

**Discussion by Hifumindesu** [ppy/osu#23173] [Ideas]
Currently, the catcher stops moving when both left and right direction keys are pressed. This can sometimes lead to slight stutters in movement especially in fast zig-zag patterns.

It'd be great if the game only used the most recently pressed direction input. (Or perhaps have a toggle in the rulesets category)

**Comment by peppy** [comment]
To confirm, this isn't how stable works and you're looking for a new behaviour here?

**Reply by Hifumindesu** [reply]
Yes, what I've described isn't currently present in stable.

stable:

https://user-images.githubusercontent.com/40629881/231670923-483cdf27-c3b1-4404-b807-84141116881b.mp4

lazer:

https://user-images.githubusercontent.com/40629881/231671011-92860362-b7a3-4765-ba2f-640b5cd7762d.mp4
